### ISSUE TITLE
App Engine Java 11 support

### ISSUE THREAD

**Issue by saturnism** [GoogleCloudPlatform/cloud-code-intellij#2665]
(Please ensure you are running the latest version of Cloud Cloud for IntelliJ with _Help > Check for Updates_.)

- Cloud Code for IntelliJ version: IntelliJ Ultimate
- OS: macOS Mojave

**What did you do?**

Would like to deploy a JAR-packaged up into App Engine Java 11

**What did you expect to see?**

In Cloud Code, options to automatically configure, add app.yaml, and configure plugins.

**What did you see instead?**

Popup to deploy to App Engine Flexible.

**Comment by saturnism** [comment]
FWIW, since the deployment path is the same w/ `gcloud app deploy`, the only thing different is to generate app.yaml for standard, that has `runtime: java11` as oppose to `flexible` snippet.

**Comment by etanshaul** [comment]
yep. I'd love to get this in. I'll keep you updated on the plan. As you said, it _shouldn't_ be much work, but I have yet to fully investigate.

@patflynn @russwolf fyi

**Comment by patflynn** [comment]
@loosebazooka who I think made changes to maven/gradle to get this to work

**Comment by loosebazooka** [comment]
I think a lot of this is handled by appengien plugins core. Lemme find the exact Gradle/maven side changes I made though

**Comment by jmah-cc** [comment]
We're prioritizing other features over AppEngine currently. I've added this to our backlog and keeping this issue open so other users can upvote this.

**Comment by jmah-cc** [comment]
Won't fix, not a priority at this time. Please upvote if you'd like to see this improvement and we'll re-visit

**Comment by tjhasan** [comment]
Hello. It's been 4 years since this issue was opened, and 2 years since it was closed. A lot has happened since then, and one of those things is an announcement for [end of support for Java 8 in App Engine Standard on January of 2024](https://cloud.google.com/appengine/docs/standard/lifecycle/support-schedule) (less than 1 year from now). So it would be pretty cool if this was given some priority and implemented now.

**Comment by ruomengz** [comment]
We are currently in the process of prioritizing this work, we will update in this thread. Thanks!

**Comment by tjhasan** [comment]
Hello! We're getting pretty close to the deadline now. Any updates on this?

**Comment by ivanporty** [comment]
There are no plans to support this as of now

**Comment by tjhasan** [comment]
I see. So, generally speaking, how should developers using Google App Engine Standard go about debugging and seamlessly deploying to their GAE Standard instances after the Java 11 requirement in late January? 

It sounds like this extension, as well as the extension for Eclipse are just going to stop working after January 2024. [The documentation for intellij](https://cloud.google.com/code/docs/intellij/create-standard) and for [Eclipse ](https://cloud.google.com/eclipse/docs/creating-new-webapp) will become pointless. 

Are developers expected to debug using logs and snapshot debuggers like it's 2003 again. It seems like the lives of people who develop on Google App Engine Standard will become significantly harder after January 2024 because Google isn't updating their plugins to match with their deprecation that was planned almost a year in advance.

**Comment by francescoare** [comment]
Please provide us a simple way to migrate our application from Java 8 to Java 11 using Eclipse. We have a very old project (> 10 years) that is still running well. Starting from january 2024 we will not be able to update!!!!!"

**Comment by francescoare** [comment]
> There are no plans to support this as of now

Why are we still lacking simple instructions for migrating to Java 11 in Eclipse, especially with the January 2024 deadline looming so close? It's urgent that we get clear, step-by-step guidance immediately!

**Comment by vbresan** [comment]
[Java 11/17 migration documentation](https://cloud.google.com/appengine/migration-center/standard/migrate-to-second-gen/java-differences). Recommended [Maven](https://cloud.google.com/appengine/docs/legacy/standard/java/using-maven) and [Gradle](https://cloud.google.com/appengine/docs/legacy/standard/java/tools/migrate-gradle) plugins.

**Comment by tjhasan** [comment]
Welp deadline is less than 30 days away. Looks like this plugin and the eclipse plugin will no longer have any purpose starting next month. Hopefully this work is prioritized after the fact once the writing is on the wall that just telling developers to "use maven or gradle" isn't enough. Because, let's be honest, using Maven or Gradle has ALWAYS been an option. Devs used this plugin because it made things easier, not because they weren't aware that maven or gradle was an option.

If that's the answer to the issue of Java 11 support, then why even build this plugin at all? Should have just told all the GAE Java 8 devs to use Maven or Gradle from the beginning.
